File: Backend/Models/Resource.cs
namespace PDPWebsite.Models;

public record Resource(Guid? Id, Guid CategoryId, Guid? TierId, string HtmlContent, string MarkdownContent, string PageName, ulong WriterId, bool Published)
{
    public Category Category { get; set; }
    public Tier? Tier { get; set; }
    public List<ResourceFile> Files { get; set; }
}

public record Category(Guid? Id, string Name, string Description, string IconUrl, string Path, bool HasTiers)
{
    public List<Resource> Resources { get; set; }
    public List<Tier> Tiers { get; set; }
}

public record Tier(Guid? Id, Guid CategoryId, string Name, string IconUrl, string Path)
{
    public List<Resource> Resources { get; set; }
    public Category Category { get; set; }
}

public record ResourceFile(Guid? Id, Guid ResourceId, string Name, string Path)
{
    public Resource Resource { get; set; }
}

File: Frontend/src/structs/resource.ts
export interface Resource {
    publish?: boolean;
    published?: boolean;
    id?: string;
    pageName?: string;
    htmlContent?: string;
    markdownContent?: string;
    categoryId?: string;
    category?: Category;
    tierId?: string;
    tier?: Tier;
    hostId?: string;
}

export interface Category {
    id: string;
    name: string;
    description: string;
    iconUrl: string;
    hasTiers: boolean;
    path: string;
}

export interface Tier {
    id: string;
    category: Category;
    categoryId: string;
    name: string;
    iconUrl: string;
    path: string;
}

export interface ResourceFile {
    id: string;
    name: string;
    path: string;
}